q2_ms/utils.py

```diff
@@ -0,0 +1,61 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2025, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+import subprocess
+
+EXTERNAL_CMD_WARNING = (
+    "Running external command line application(s). "
+    "This may print messages to stdout and/or stderr.\n"
+    "The command(s) being run are below. These commands "
+    "cannot be manually re-run as they will depend on "
+    "temporary files that no longer exist."
+)
+
+
+def run_command(cmd, cwd, verbose=True, env=None):
+    if verbose:
+        print(EXTERNAL_CMD_WARNING)
+        print("\nCommand:", end=" ")
+        print(" ".join(cmd), end="\n\n")
+    subprocess.run(cmd, check=True, cwd=cwd, env=env)
+
+
+def run_r_script(params, script_name, package_name):
```

Comment:
Just being a bit nit-picky, but could you perhaps swap `params` with `script_name`? That would feel a bit more natural as the command below is constructed in this order.